strutil/quantity/quantity.go

```diff
@@ -200,3 +201,239 @@ func FormatDuration(dt float64) string {
 
 	return FormatAmount(uint64(dt), 4) + years
 }
+
+// Duration is represented in nano seconds.
+type Duration uint64
+
+const (
+	NSec Duration = 1
+	USec Duration = 1000
+	MSec Duration = 1000000
+	Sec  Duration = 1000000000
+	Min  Duration = (60 * Sec)
+	Hour Duration = (60 * Min)
+	Day  Duration = (24 * Hour)
+	Year Duration = (1461 * Day / 4)
+)
+
+// Maximum number of places (units of time) to render starting with years.
+// In the Compact case, only the first two places will be rendered,
+// discarding the rest. This allows for a more predictable width (at the
+// expense of accuracy).
+type Places uint32
+
+const (
+	ShowAll  Places = 8
+	ShowComp Places = 2
+)
+
+// The minimum Duration parameter allows the rendering to discard the units
+// of time less than the minimum. However, depending on what we are
+// rendering, we either need to peform a ceiling, floor or rounding
+// operation with the remainder below the minimum.
+type RenderMode uint32
+
+const (
+	TimeLeft    RenderMode = 0
+	TimePassed  RenderMode = 1
+	TimeRounded RenderMode = 2
+)
+
+// FormatDurationGeneric formats time duration similar to systemd
+// format_timespan, but with options allowing for a more compact
+// arrangement, including shortened suffixes. The function is provided
+// with dt in seconds (as in the output of time.Now().Seconds()). The 'min'
+// and 'max' Duration allows limiting the unit range rendered. Place values
+// less than 'min' will be processed according to RenderMode. RenderMode
+// allows rounding, flooring or ceiling to be applied to place value below
+// 'min'. Durations above the 'max' place value will result in "ages!".
+func FormatDurationGeneric(dt float64, min Duration, max Duration, count Places, mode RenderMode, space bool) string {
```

Comment:
I prefer flags or something equivalent over booleans in an API. I find it hard to reason about something like:
```
FormatDuration(..., false)
```
there one has to go to the api docs to see what the "false" means. By contrast using flags like:
```
FormatDuration(..., IncludeSpace)
```
does not requires this (fwiw `IncudeSpace` is just a strawman that came to my mind first, there are probably better names for this flag).

Reply:
Fixed. SpaceOn / SpaceOff.